### PR TITLE
data_factory_linked_service_sql_server: support `key_vault_password`

### LINKED ISSUE #6481: Adding a keyvault feature in  azure datafactory linked service
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I have found some configuration is missing in datafactory_linkedservice .As in terraform only connection string variable is present but in azure you can see two options one is connection string and other one is AKV so we as a team decided there should be AKV(Azure keyvault) variable embedded in the module.Because adding only connection string actually gives less value because all the users who have access to datafactory can have ability to get the connection string .So need the community help to add this feature .

### New or Affected Resource(s)

azurerm_data_factory_linked_service_sql_server

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

### PR DESCRIPTION
**Pull request by allantargino**
fixes #6481

Following the same interface introduced at #9928, where we added `key_vault_password` property on `azurerm_data_factory_linked_service_synapse` to support passwords stored in Key Vault secret through an ADF linked service, this PR adds the same capability to `azurerm_data_factory_linked_service_sql_server`.

The following example

```hcl
resource "azurerm_data_factory_linked_service_sql_server" "test" {
  name                = "SqlServerLinkedSvc"
  resource_group_name = azurerm_resource_group.test.name
  data_factory_name   = azurerm_data_factory.test.name
  connection_string = "Integrated Security=False;Data Source=test;Initial Catalog=test;User ID=test;"
  key_vault_password {
    linked_service_name = "KeyVaultLinkedSvc"
    secret_name         = "MySecret"
  }
}
```

produces the following Linked Service on ADF:

```json
{
    "name": "SqlServerLinkedSvc",
    "type": "Microsoft.DataFactory/factories/linkedservices",
    "properties": {
        "annotations": [],
        "type": "SqlServer",
        "typeProperties": {
            "connectionString": "Integrated Security=False;Data Source=test;Initial Catalog=test;User ID=test;",
            "password": {
                "type": "AzureKeyVaultSecret",
                "store": {
                    "referenceName": "KeyVaultLinkedSvc",
                    "type": "LinkedServiceReference"
                },
                "secretName": "MySecret"
            }
        }
    }
}
```

### CODE REVIEW

File: azurerm/internal/services/datafactory/data_factory_linked_service_sql_server_resource.go [hunk 1]
```diff
@@ -223,18 +244,22 @@ func resourceArmDataFactoryLinkedServiceSQLServerRead(d *schema.ResourceData, me
...
-				connectionString = val
+			if val, ok := properties.ConnectionString.(string); ok {
+				d.Set("connection_string", val)
 			} else {
+				d.Set("connection_string", "")
 				log.Printf("[DEBUG] Skipping connection string %q since it's not a string", val)
 			}
 		}
+
+		if keyVaultPassword, ok := properties.Password.AsAzureKeyVaultSecretReference(); ok {
```
Comment (katbyte) [inline azurerm/internal/services/datafactory/data_factory_linked_service_sql_server_resource.go:257]: We'll need to nic check password here

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @allantargino - one small comment i've left inline and this should be good

File: azurerm/internal/services/datafactory/data_factory_linked_service_sql_server_resource.go (hunk 1 above)
Comment (allantargino) [reply azurerm/internal/services/datafactory/data_factory_linked_service_sql_server_resource.go]: Done @katbyte

**Comment by allantargino** [comment]
Hi @katbyte, following the refactor from #10114, I refactored this PR and solved conflicts with master.

**Review by katbyte — APPROVED** [review]
Thanks @allantargino - LGTM 👍

**Pull request merged**